src/cost_functions.jl

```diff
@@ -1,3 +1,15 @@
+export DECostFunction, CostVData, L2Loss, Regularization
+
+type Regularization{L,P} <: DECostFunction
+  λ::L
+  penalty::P
+end
+
+function (f::Regularization)(p)
```

Comment:
MahalanobisPenalty works on arrays so if the parameter estimation problem is to find a single parameter, it spits out the error. Any workaround?